x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/SSLService.java

```diff
@@ -826,11 +828,21 @@ private static String sslContextAlgorithm(List<String> supportedProtocols) {
     }
 
     private boolean shouldEnableDiagnoseTrust() {
-        if (XPackSettings.FIPS_MODE_ENABLED.get(settings) && DIAGNOSE_TRUST_EXCEPTIONS_SETTING.exists(settings) == false ) {
+        // We disable the DiagnosticTrustManager in Java 8 when SunJSSE is set in FIPS 140 mode, as it doesn't allow X509TrustManager to be
+        // wrapped
+        if (inSunJsseInFipsMode()) {
```

Comment:
@tvernum I asked for your review since this is changing this default behavior in a patch release ( 7.7.2 ): 

- Before the change, if a user configured a JDK8 JVM using the SunJSSE provider in fips mode we would allow them to explicitly enable the diagnostic trust manager on startup and we would fail in runtime
- After the change, we do not allow the user to explicitly enable the diagnostic trust manager in this scenario at all and we print a message on INFO level.

I think this is fine, because the new behavior is much more sane and I could argue is a bugfix over the old one, but I wanted to get a +1

Reply:
ping @tvernum for a quick +1/-1

Reply:
+1